whisper/whisperv6/peer.go

```diff
@@ -36,6 +36,7 @@ type Peer struct {
 
 	trusted        bool
 	powRequirement float64
+	bloomFilter    []byte // may contain nil in case of full node
```

Comment:
I'm not a fan of having `nil` all over the place, especially in things that amount to references. Old C reflex I guess. Wouldn't it make sense to have `bloomFilter` to be all `1`s if this is a full node? Looking at the code, I see no reason to separate the case of a full node from the case where the some stuff is filtered. The full node case is essentially: filter nothing, and making the distinction is just adding confusion by adding more `if bloomFilter == nil`s that is needed.

Reply:
i thought about productivity: most nodes will be full nodes, so comparing 64 bytes in a cycle would be an overkill. but i have to admit that you are absolutely right -- readability (and good engineering practices in gerenral) have higher priority over productivity. 

however, my first priority today is to deliver the minimum viable product of v6. so, you are welcome to do the necessary refactoring after v6 is realeased.